addons/mail/static/src/core/messaging_service.js

```diff
@@ -191,20 +191,29 @@ export class Messaging {
                     break;
                 case "discuss.channel/leave":
                     {
-                        const thread = this.threadService.insert({
-                            ...notif.payload,
-                            model: "discuss.channel",
-                        });
-                        this.threadService.remove(thread);
-                        if (thread.localId === this.store.discuss.threadLocalId) {
-                            this.store.discuss.threadLocalId = undefined;
-                        }
+                        const thread = this._removeThread(notif.payload);
                         this.notificationService.add(
                             sprintf(_t("You unsubscribed from %s."), thread.displayName),
                             { type: "info" }
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this._removeThread({ id: notif.payload.id });
+                        this.store.discuss.starred.messages = this.store.discuss.starred.messages.filter(
+                            (message) => message.resId !== thread.id
+                        );
+                        this.store.discuss.starred.counter = this.store.discuss.starred.messages.length;
+                        this.store.discuss.inbox.messages = this.store.discuss.inbox.messages.filter(
+                            (message) => message.resId !== thread.id
+                        );
```

Comment:
A thread is not uniquely identified by `id`. You should check `thread.id` and `thread.model`.

For example, if you have 2 threads `discuss.channel,1` and `project.task,1` and `discuss.channel,1` is deleted, you don't want to remove `project.task,1` messages from inbox